community/modules/compute/schedmd-slurm-gcp-v6-partition/metadata.yaml

```diff
@@ -0,0 +1,2 @@
+ghpc:
+  inject_module_id: partition_name
```

Comment:
I think this deserves a design doc. 
I like what you have done here, but injecting a varialbe "module_id" would also be very simple and transparent, since you then do this logic in terraform. I would be curious to know what others think. 
for instance: 
you could define a variable "module_id" that when set, ghpc will push the module id. Then in the main.tf, if partition_name == null and module_id != null, then partition_name = module_id.

In my opinion it would be preferrable as it would still allow this module to work without special ghpc behavior. 

On the other hand, what is nice here is the ability to tell ghpc to rename variables, meaning you would perhaps no longer need simple wrappers: one could have more or less complex metadata files that would instruct hpc how to map various variables... which is also powerful.